src/util/hb_arena.c

```diff
@@ -5,58 +5,149 @@
 #include "../include/util/hb_arena.h"
 
 #include <stdbool.h>
+#include <stdint.h>
+#include <string.h>
 #include <sys/mman.h>
 
 #define KB(kb) (1024 * kb)
 #define MB(mb) (1024 * KB(mb))
+#define HB_ARENA_PAGE_SIZE_DEFAULT KB(4)
 
-bool hb_arena_init(hb_arena_T* allocator, size_t size) {
-  allocator->memory = mmap(NULL, size, PROT_READ | PROT_WRITE, MAP_PRIVATE | MAP_ANONYMOUS, -1, 0);
+bool hb_arena_init(hb_arena_T* allocator, size_t initial_size) {
+  if (initial_size == 0) { initial_size = HB_ARENA_PAGE_SIZE_DEFAULT; }
```

Comment:
Suggestion: Remove default.

Making people think about what their initial size should be, may be advantageuous here. 

Additionally it just looks weird to init an arena with an initial size 0>.